code/datums/diseases/_disease.dm

```diff
@@ -155,7 +155,7 @@ var/list/diseases = subtypesof(/datum/disease)
 		if(disease_flags & CAN_RESIST)
 			if(!(type in affected_mob.resistances))
 				affected_mob.resistances += type
-				remove_virus()
+		remove_virus()
```

Comment:
I know this is an ultra one-lear late follow up @phil235 But you're, indeed, correct here.

What can happen is if you have a disease that cannot be resisted and it's cured, it will fail to garbage collect, since the virus is still technically in the mob's virus list...then it'll just be hard deleted and leave a null entry in the `viruses` list.

You did the right thing here.